### ISSUE TITLE
Bug when adding multiple keys to input map

### ISSUE THREAD

**Issue by IndieQuest** [godotengine/godot#44155]
**Godot version:**
v4.0.dev.custom_build.93b107ac9

**OS/device including version:**
Windows 10, Nvidia 1070

**Issue description:**
When adding multiple keys in succession a new input binding is created with the name of the last entered key ([see video](https://github.com/godotengine/godot/files/5650939/InputMapBug.zip)).

**Steps to reproduce:**
1. Open project settings
2. Go to the input map tab
3. Create a new action
4. Add **A** as a key for the new action
5. Without pressing anything else, add the **S** key to the same action

 

**Minimal reproduction project:**
[Test.zip](https://github.com/godotengine/godot/files/5650961/Test.zip)

**Comment by IndieQuest** [comment]
Just saw that the video I attached is corrupted.
Adding an new one with some more examples and behaviors regarding this issue.
BTW, I could not find the relevant code that handles input mapping but if you could point me to it I would love to take a swing at fixing the issue.
[InputMapBug.zip](https://github.com/godotengine/godot/files/5651115/InputMapBug.zip)

**Comment by akien-mga** [comment]
The relevant code should be in `editor/input_map_editor.cpp`.

**Comment by IndieQuest** [comment]
I am able to conclude that in some cases the **TreeItem** sent to [this function](https://github.com/godotengine/godot/blob/master/editor/input_map_editor.cpp#L562) is referencing the first child of the actual action instead of the action itself.
This is the first time I am working with the editor code so I am not yet familiar with how everything works, will continue working on this tommorow.

**Comment by EricEzaM** [comment]
That is pretty weird - I have worked quite a bit with the editor code and just a warning, this may not be trivial to solve.

FYI I am in the middle of a bit of a rewrite of the input map editor, I actually remove input_map_editor.cpp and replace with with another, more generic implementation that is not dependant on Project Settings and thus could be used elsewhere if required in the future.

Here is a preview of what it looks like. Since I did a complete rewrite this bug does not seems to be there.

![inputmap_demo](https://user-images.githubusercontent.com/41730826/101416999-adb0e280-3936-11eb-941f-56da3a219711.gif)
